src/Persistence/Yaml/NamedObjectsCollectionConverter.cs

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using YamlDotNet.Core;
+using YamlDotNet.Core.Events;
+using YamlDotNet.Serialization;
+using YamlDotNet.Serialization.Utilities;
+
+namespace Microsoft.PowerPlatform.PowerApps.Persistence.Yaml;
+
+public class NamedObjectsCollectionConverter<T> : IYamlTypeConverter
+    where T : class, INamedObject
+{
+    public required YamlSerializationOptions Options { get; set; }
+
+    public IValueDeserializer? ValueDeserializer { get; set; }
+
+    public IValueSerializer? ValueSerializer { get; set; }
+
+    public bool Accepts(Type type)
+    {
+        return
+            type == typeof(List<T>) || type.IsSubclassOf(typeof(List<T>)) ||
+            type == typeof(IList<T>) || type.IsSubclassOf(typeof(IList<T>)) ||
```

Comment:
I suspect that `IsSubclassOf(typeof(IList<T>))` is not really doing much. Being `IList` an interface, it should _probably_ be `IsAssignableFrom`

Reply:
Interfaces behave differently. It works with this logic. If you know how to improve let's change it